### ISSUE TITLE
Evaluated type of expression varies based on evaluation order

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2983]
This is a tracking bug. The original is found here: https://github.com/microsoft/pylance-release/issues/2324.

The evaluated type of the return expression depends on the order of evaluation. It should be order invariant.

To repro, hover over the `e` in the return statement while modifying the file. The hover will cause the `e` in the return statement to be evaluated prior to the other statements, and the inferred return type will be evaluated as `Literal[0]` rather than the correct answer of `Unknown | Literal[0]`.

```python
def f(x):
    e = 0
    for _ in [0]:
        e += x
    return e
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.228, which I just published. It will also be included in the next release of pylance.

**Comment by erictraut** [comment]
Reopening for investigation because user reports that it's still a problem.

**Comment by erictraut** [comment]
This is included in pyright 1.1.233, which I just published. It will also be included in the next release of pylance.
